src/chainlit/frontend/src/components/chat/message/inlined.tsx

```diff
@@ -20,19 +16,40 @@ export default function InlinedElements({ elements, actions }: Props) {
     return null;
   }
 
-  const images = elements.filter(
-    (el) => el.type === ElementType.img
-  ) as IImageElement[];
-
-  const texts = elements.filter(
-    (el) => el.type === ElementType.txt
-  ) as ITextElement[];
+  // Categorize the elements by element type
+  // The TypeScript dance is needed to make sure we can do elementsByType.image
+  // and get an array of IImageElement
+  const elementsByType = elements.reduce(
+    (acc, el: AllElements) => {
+      if (!acc[el.type]) {
+        acc[el.type] = [];
+      }
+      const array = acc[el.type] as Extract<
+        AllElements,
+        { type: typeof el.type }
+      >[];
+      array.push(el);
+      return acc;
+    },
+    {} as {
+      [K in ElementType]: Extract<AllElements, { type: K }>[];
+    }
+  );
 
   return (
     <Stack gap={1} mt={1}>
-      {images.length ? <InlinedImageList images={images} /> : null}
-      {Object.keys(texts).length ? <InlinedTextList items={texts} /> : null}
+      {elementsByType.image?.length ? (
+        <InlinedImageList
+          items={elementsByType.image.filter((x) => x.type == 'image')}
```

Comment:
since we have elementsByType.image do we still need to filter again?

Reply:
Right, this was part of a fight I had with TypeScript, it's removed now!